tests/MongoDB.Driver.Tests/UnifiedTestOperations/UnifiedEntityMap.cs

```diff
@@ -0,0 +1,526 @@
+        {
+            _changeStreams.Add(changeStreamId, changeStream);
+        }
+
+        public void AddResult(string resultId, BsonValue value)
+        {
+            _results.Add(resultId, value);
+        }
+
+        public void Dispose()
+        {
+            if (_changeStreams != null)
+            {
+                foreach (var changeStream in _changeStreams.Values)
+                {
+                    changeStream?.Dispose();
+                }
+            }
+            if (_sessions != null)
+            {
+                foreach (var session in _sessions.Values)
+                {
+                    session?.Dispose();
+                }
+            }
+            if (_clients != null)
+            {
+                foreach (var client in _clients.Values)
+                {
+                    client?.Dispose();
+                }
+            }
+        }
+
+        public IGridFSBucket GetBucket(string bucketId)
+        {
+            return _buckets[bucketId];
+        }
+
+        public IEnumerator<ChangeStreamDocument<BsonDocument>> GetChangeStream(string changeStreamId)
+        {
+            return _changeStreams[changeStreamId];
+        }
+
+        public IMongoClient GetClient(string clientId)
+        {
+            return _clients[clientId];
+        }
+
+        public IMongoCollection<BsonDocument> GetCollection(string collectionId)
+        {
+            return _collections[collectionId];
+        }
+
+        public IMongoDatabase GetDatabase(string databaseId)
+        {
+            return _databases[databaseId];
+        }
+
+        public EventCapturer GetEventCapturer(string clientId)
+        {
+            return _clientEventCapturers[clientId];
+        }
+
+        public BsonValue GetResult(string resultId)
+        {
+            return _results[resultId];
+        }
+
+        public IClientSessionHandle GetSession(string sessionId)
+        {
+            return _sessions[sessionId];
+        }
+
+        public BsonDocument GetSessionId(string sessionId)
+        {
+            return _sessionIds[sessionId];
+        }
+
+        public bool HasBucket(string bucketId)
+        {
+            return _buckets.ContainsKey(bucketId);
+        }
+
+        public bool HasChangeStream(string changeStreamId)
+        {
+            return _changeStreams.ContainsKey(changeStreamId);
+        }
+
+        public bool HasClient(string clientId)
+        {
+            return _clients.ContainsKey(clientId);
+        }
+
+        public bool HasCollection(string collectionId)
+        {
+            return _collections.ContainsKey(collectionId);
+        }
+
+        public bool HasDatabase(string databaseId)
+        {
+            return _databases.ContainsKey(databaseId);
+        }
+
+        public bool HasSession(string sessionId)
+        {
+            return _sessions.ContainsKey(sessionId);
+        }
+    }
+
+    public class UnifiedEntityMapBuilder
```

Comment:
This looks great.

Not only have you moved the parsing code out of the the `UnifiedEntityMap` constructor, you have moved it to an entirely different class, which makes tons of sense.

Reply:
Thanks!